pkg/services/accesscontrol/resourcepermissions/api.go

```diff
@@ -42,21 +43,27 @@ func (a *api) registerEndpoints() {
 		licenseMW = nopMiddleware
 	}
 
+	teamUIDResolver := team.MiddlewareTeamUIDResolver(a.service.teamService, ":teamID")
+	teamUIDResolverResource := func() web.Handler { return func(c *contextmodel.ReqContext) {} }() // no-op
+	if a.service.options.Resource == "teams" {
+		teamUIDResolverResource = team.MiddlewareTeamUIDResolver(a.service.teamService, ":resourceID")
```

Comment:
when porting extra resources like users to UIDs we should make this resolver an option